engine.js

```diff
@@ -36,55 +36,119 @@ var filterSubject = function(subject) {
 // This can be any kind of SystemJS compatible module.
 // We use Commonjs here, but ES6 or AMD would do just
 // fine.
-module.exports = function(options) {
+module.exports = function(options, inquirer) {
   var types = options.types;
-
-  var length = longest(Object.keys(types)).length + 1;
-  var choices = map(types, function(type, key) {
+  var longestTypeChoice = longest(Object.keys(types)).length + 1;
+  var typeChoices = map(types, function(type, key) {
     return {
-      name: (key + ':').padEnd(length) + ' ' + type.description,
+      name: (key + ':').padEnd(longestTypeChoice) + ' ' + type.description,
       value: key
     };
   });
 
+  var predefinedScopes = options.scopes || [];
+  var hasPredefinedScopes = predefinedScopes.length > 0;
+
+  var otherScopeChoice = 'something else...';
+  var scopeChoices = [{ name: chalk.dim(chalk.white('(skip)')), value: '' }]
+    .concat(
+      map(predefinedScopes, function(scope) {
+        return {
+          name: scope,
+          value: scope
+        };
+      })
+    )
+    .concat([
+      new inquirer.Separator(''),
+      { name: otherScopeChoice, value: otherScopeChoice }
+    ]);
+
+  var defaultScopeChoice;
+
+  if (options.defaultScope) {
+    var foundDefault = scopeChoices.findIndex(function(choice) {
+      if (choice.type === 'separator') {
+        return false;
+      }
+      return (
+        choice.value === options.defaultScope ||
+        choice.name === options.defaultScope
+      );
+    });
+
+    defaultScopeChoice =
+      foundDefault === -1 ? scopeChoices.length - 1 : foundDefault;
+  }
+
   return {
     // When a user runs `git cz`, prompter will
-    // be executed. We pass you cz, which currently
-    // is just an instance of inquirer.js. Using
-    // this you can ask questions and get answers.
+    // be executed.
     //
     // The commit callback should be executed when
     // you're ready to send back a commit template
     // to git.
     //
     // By default, we'll de-indent your commit
     // template and will keep empty lines.
-    prompter: function(cz, commit) {
+    prompter: function(commit) {
```

Comment:
We may be able to simplify the engine even more by just having it directly run `cz.prompt` / `inquirer.prompt`, but doing such a change right now would result in the changes due to predefined scopes being obscured. THis future change would look like:

```
module.exports = function(options, inquirer, commit) {
  // ...
  inquirer.prompt().then(...)
}
```

Without returning a "prompter" method here.

Reply:
Essentially I'm trying to make this easier to review by avoiding changes that would make additional lines of code look changed when they've not actually changed, other than an indentation level change.